utils.py

```diff
@@ -147,27 +145,130 @@ def process_what_to_run_randoms(pairs_to_test, random_counterpart):
   """
-  result_summary = {'random': []}
+  # print class, it will be the same for all
+  print("Class =", results[0]['target_class'])
+
+  # prepare data
+  # dict with keys of concepts containing dict with bottlenecks
+  result_summary = {}
+    
+  # random
+  random_i_ups = {}
+    
   for result in results:
-    if 'random' in result['cav_concept']:
-      result_summary['random'].append(result)
-    else:
-      if result['cav_concept'] not in result_summary:
-        result_summary[result['cav_concept']] = []
-      result_summary[result['cav_concept']].append(result)
-  random_i_ups = [item['i_up'] for item in result_summary['random']]
-
+    if result['cav_concept'] not in result_summary:
+      result_summary[result['cav_concept']] = {}
+    
+    if result['bottleneck'] not in result_summary[result['cav_concept']]:
+      result_summary[result['cav_concept']][result['bottleneck']] = []
+    
+    result_summary[result['cav_concept']][result['bottleneck']].append(result)
+    
+    # store random
+    # check if random concepts are from a supplied list first
+    if (random_concepts and result['cav_concept'] in random_concepts) \
+        or (not random_concepts and 'random500_' in result['cav_concept']):
+      if result['bottleneck'] not in random_i_ups:
+        random_i_ups[result['bottleneck']] = []
+        
+      random_i_ups[result['bottleneck']].append(result['i_up'])
+    
+  # to plot, must massage data again 
+  plot_data = {}
+    
+  # print concepts and classes with indentation
   for concept in result_summary:
-    if 'random' is not concept:
-      i_ups = [item['i_up'] for item in result_summary[concept]]
-      print('%s: TCAV score: %.2f (+- %.2f), random was %.2f' % (
-          concept, np.mean(i_ups), np.std(i_ups), np.mean(random_i_ups)))
+        
+    # if not random
+    if (random_concepts and concept not in random_concepts) \
+        or (not random_concepts and 'random500_' not in concept):
+      print(" ", "Concept =", concept)
+
+      for bottleneck in result_summary[concept]:
+        i_ups = [item['i_up'] for item in result_summary[concept][bottleneck]]
+        
+        # Calculate statistical significance
+        _, p_val = ttest_ind(random_i_ups[bottleneck], i_ups)
+                  
+        if bottleneck not in plot_data:
+          plot_data[bottleneck] = {'bn_vals': [], 'bn_stds': [], 'significant': []}
+
+        if p_val > min_p_val:
+          # statistically insignificant
+          plot_data[bottleneck]['bn_vals'].append(0.01)
+          plot_data[bottleneck]['bn_stds'].append(0)
+          plot_data[bottleneck]['significant'].append(False)
+            
+        else:
+          plot_data[bottleneck]['bn_vals'].append(np.mean(i_ups))
+          plot_data[bottleneck]['bn_stds'].append(np.std(i_ups))
+          plot_data[bottleneck]['significant'].append(True)
+
+        print(3 * " ", "Bottleneck =", ("%s. TCAV Score = %.2f (+- %.2f), "
```

Comment:
can you also add a string indicator to these results saying "significant" or "not significant" in addition to the calculated p value?